lib/util/SparqlExpressionEvaluator.js

```diff
@@ -32,34 +59,53 @@ SparqlExpressionEvaluator.evaluate = function (expression, bindings) {
   return new SparqlExpressionEvaluator(expression)(bindings);
 };
 
-// The null operation
-function noop() { }
-
 // Evaluators for each of the expression types
 evaluators = {
   // Does nothing
-  null: function () { return noop; },
+  null: function () {
+    return noop;
+  },
 
   // Evaluates an IRI, literal, or variable
   string: function (expression) {
     // Evaluate a IRIs or literal to its own value
-    if (expression[0] !== '?')
-      return function () { return expression; };
+    if (expression[0] !== '?') {
+      return function () {
+        return expression;
+      };
+    }
     // Evaluate a variable to its value
-    else
-      return function (bindings) { return bindings && bindings[expression]; };
+    else {
+      return function (bindings) {
+        if (!bindings || !(expression in bindings))
+          throw new Error('Cannot evaluate variable ' + expression + ' because it is not bound.');
+        return bindings[expression];
+      };
+    }
   },
 
   // Evaluates an operation
   operation: function (expression) {
     // Find the operator and check the number of arguments matches the expression
-    var operatorName = expression.operator || expression.function,
+    var operatorName = expression.operator,
         operator = operators[operatorName];
     if (!operator)
-      throw new UnsupportedOperatorError(operatorName);
-    if (operator.length !== expression.args.length)
-      throw new InvalidArgumentsNumberError(operatorName, expression.args.length, operator.length);
-
+      throw new Error('Unsupported operator: ' + operatorName.toUpperCase() + '.');
+    if (operator.argBounds) {
+      var min = operator.argBounds.min, max = operator.argBounds.max;
+      if ((min && max && (min > expression.args.length || max < expression.args.length)) ||
+        (min && (min > expression.args.length)) ||
+        (max && (max < expression.args.length))) {
```

Comment:
Good that we have bounds checking.

Could it be that `max` is set to `0`, i.e., no arguments? That would fail the bound check. E.g., `UUID` and `strUUID`, if we pass them arguments, I think this check would not work.